lib/src/operance_data_column.dart

```diff
@@ -50,8 +56,81 @@ class OperanceDataColumn<T> {
   /// Defaults to `false`.
   final bool sortable;
 
+  /// Indicates whether the column is filterable.
+  ///
+  /// Defaults to `false`.
+  final bool filterable;
+
   /// The width of the column.
   ///
   /// Defaults to an instance of [OperanceDataColumnWidth].
   final OperanceDataColumnWidth width;
+
+  /// A custom comparator function for sorting.
+  ///
+  /// If provided, this function will be used for sorting instead of the default
+  /// comparison. The function should return a negative value if a < b, zero if
+  /// a == b, and a positive value if a > b.
+  final int Function(T a, T b)? comparator;
```

Comment:
nice! was thinking about local sorting, how will this affect backend sorting though?